### PR TITLE
Fix black ox brew crash

### PR DESCRIPTION
**Pull request by JoeyBG**
- [x] Light brewing talent CRD logic was reversed for Celestial Brew and Purifying Brew
- [x] Converted BlackOxBrew analyzer to Typescript
- [x] Fixed object division when computing threshold, resulting in `NaN` (thanks @emallson!)
- [x] Fixed crash when BOB was cast with CB available (see https://github.com/WoWAnalyzer/WoWAnalyzer/pull/4280/files#diff-2d232e67a4012b33427cbab6f6cb022669bed52e8be098612ed5efc20c25b342R88).

### CODE REVIEW

**Comment by JoeyBG** [comment]
Ugh. I will add the changelog entry once/if #4279 is merged :)

File: src/parser/monk/brewmaster/modules/Abilities.ts
Comment (emallson) [inline src/parser/monk/brewmaster/modules/Abilities.ts:122]: good catch

**Review by emallson — APPROVED** [review]

**Comment by emallson** [comment]
Since you need to add a changelog entry anyway, i'll let you resolve the merge conflict in `Abilities.ts`

**Comment by JoeyBG** [comment]
I added the changelog. I think the CD changes broken the integration tests... What is the best way to update them, manually? I'm not too familiar with that test engine.

**Comment by Sharrq** [comment]
`yarn test` 

Then `u` to update failing snapshots.

**Pull request merged**
